packages/test/test-end-to-end-tests/src/test/attachRegisterLocalApiTests.spec.ts

```diff
@@ -17,10 +17,25 @@ import {
 	ITestObjectProvider,
 } from "@fluidframework/test-utils";
 import { SharedObject } from "@fluidframework/shared-object-base";
-import { IContainerRuntimeBase } from "@fluidframework/runtime-definitions";
+import {
+	IContainerRuntimeBase,
+	type IFluidDataStoreContext,
+} from "@fluidframework/runtime-definitions";
 import type { ISharedMap } from "@fluidframework/map";
 import { describeCompat } from "@fluid-private/test-version-utils";
 
+function onAttachChange(
```

Comment:
Add a comment here that this is for back-compat. This can also be removed after 1 internal release right. Basically, we have to support 1 version back-compat between data store runtime and container runtime?

Reply:
Also, this test runs in "NoCompat" mode so these tests shouldn't be failing right?

Reply:
They do fail because test itself relies on event that is gone. Tests rely on knowing when context changes its state, but context no longer exposes anything to learn that other than calling channel API directly.

Reply:
So, it's not for back-compat, it's for spying on context where context no longer provides these capabilities.
The only other alternative I see is to get rid of these tests

Reply:
Okay, I didn't know that the tests relied on this event.